### PR TITLE
collect scanner's status when es_http_scan_node close

### LINKED ISSUE #1859: EsHttpScanNode may be blocked while start_scanners
* when open EsHttpScanNode we call start_scanners
* start_scanners wait all futures return their status,  it's a sync logic
* but open function is called in PlanFragmentExecutor before batch read
* so read will be blocked until all scanners finish

### PR DESCRIPTION
**Pull request by shgxwxl**
add collect_scanners_status function ,which need to call after start_scanners to check status

### CODE REVIEW

File: be/src/exec/es_http_scan_node.cpp [hunk 1]
```diff
@@ -176,12 +183,12 @@ Status EsHttpScanNode::get_next(RuntimeState* state, RowBatch* row_batch,
 
     if (_eos) {
         *eos = true;
-        return Status::OK();
+        return collect_scanners_status();
```
Comment (imay) [inline be/src/exec/es_http_scan_node.cpp]: You can do it in close() other than call this every time.

File: be/src/exec/es_http_scan_node.cpp [hunk 2]
```diff
@@ -153,11 +153,18 @@ Status EsHttpScanNode::start_scanners() {
...
-                    _scan_ranges.size(), std::ref(p));
+                    _scan_ranges.size(), std::ref(_scanners_status[i]));
+    }
+    return Status::OK();
+}
+
+Status EsHttpScanNode::collect_scanners_status() {
+    for (int i = 0; i < _scan_ranges.size(); i++) {
+        std::future<Status> f = _scanners_status[i].get_future();
         Status status = f.get();
```
Comment (imay) [inline be/src/exec/es_http_scan_node.cpp]: RETURN_IF_ERROR(f.get())

**Review by imay — CHANGES_REQUESTED** [review]

**Comment by shgxwxl** [comment]
fix #1859  only collect all scanners status when es_http_scan_node
make collect_scanners_status from public to private
change ut for es_http_scan_node and run ok

File: be/src/exec/es_http_scan_node.cpp [hunk 3]
```diff
@@ -266,7 +273,10 @@ Status EsHttpScanNode::close(RuntimeState* state) {
 
     _batch_queue.clear();
 
-    return ExecNode::close(state);
+    RETURN_IF_ERROR(ExecNode::close(state));
+
+    //collect scanners status at last
+    return collect_scanners_status();
```
Comment (imay) [inline be/src/exec/es_http_scan_node.cpp]: should call this function before ExecNode::close(state)

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**